android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/RepeatingGroupFactory.java

```diff
@@ -0,0 +1,327 @@
+                            String repeatingGroupKey = (String) rootLayout.getChildAt(i).getTag(R.id.repeating_group_key);
+                            keysToRemove.add(repeatingGroupKey);
+                            rootLayout.removeViewAt(i);
+                        }
+                        // remove deleted fields from form json
+                        int len = fields.length();
+                        for (int i = len - 1; i >= 0; i--) {
+                            String[] key = ((String) fields.getJSONObject(i).get(KEY)).split("_");
+                            if (keysToRemove.contains(key[key.length - 1])) {
+                                fields.remove(i);
+                            }
+                        }
+                        LinearLayout referenceLayout = (LinearLayout) ((LinearLayout) parent).getChildAt(0);
+                        referenceLayout.getChildAt(0).setTag(R.id.repeating_group_item_count, rootLayout.getChildCount());
+                    } catch (JSONException e) {
+                        Log.e(TAG, e.getStackTrace().toString());
+                    }
+                } else {
+                    for (View repeatingGroup : repeatingGroups) {
+                        rootLayout.addView(repeatingGroup);
+                    }
+                }
+
+                ((JsonApi) widgetArgs.getContext()).invokeRefreshLogic(null, false, null, null);
+                hideProgressDialog();
+                doneButton.setImageResource(R.drawable.ic_done_green);
+            }
+        }
+
+        new AttachRepeatingGroupTask().execute();
+    }
+
+    private LinearLayout buildRepeatingGroupLayout(ViewParent parent, WidgetArgs widgetArgs) throws Exception {
+        Context context = widgetArgs.getContext();
+
+        LinearLayout repeatingGroup = new LinearLayout(context);
+        repeatingGroup.setLayoutParams(WIDTH_MATCH_PARENT_HEIGHT_WRAP_CONTENT);
+        repeatingGroup.setOrientation(LinearLayout.VERTICAL);
+
+        LinearLayout rootLayout = (LinearLayout) ((LinearLayout) parent).getChildAt(0);
+        EditText referenceEditText = (EditText) rootLayout.getChildAt(0);
+        TextView repeatingGroupLabel = new TextView(context);
+        formatRepeatingGroupLabelText(referenceEditText, repeatingGroupLabel, context);
+        repeatingGroup.addView(repeatingGroupLabel);
+
+        JSONArray repeatingGroupJson = new JSONArray(repeatingGroupLayouts.get(((LinearLayout) parent).getId()));
+        String uniqueId = UUID.randomUUID().toString();
+        for (int i = 0; i < repeatingGroupJson.length(); i++) {
+            JSONObject element = repeatingGroupJson.getJSONObject(i);
+            String elementType = element.optString(TYPE, null);
+            if (elementType != null) {
+                addUniqueIdentifiers(element, uniqueId);
+                FormWidgetFactory factory = JsonFormInteractor.getInstance().map.get(elementType);
```

Comment:
Consider getting JsonFormInteractor from FormFragment getPresenter which has the reference to interactor.

With approach then if a project overrides `JsonFormInteractor` which is normally the case then `JsonFormInteractor.getInstance() `will always the parent `JsonFormInteractor` and thereby skipping custom added/overridden factories

Reply:
Done.